.circleci/config.yml

```diff
@@ -364,11 +367,73 @@ commands:
           det-version: <<parameters.det-version>>
           agent-instance-type: <<parameters.agent-instance-type>>
           max-dynamic-agents: <<parameters.max-dynamic-agents>>
-      - set-master-address:
+      - set-master-address-aws:
           cluster-id: ${CLUSTER_ID}
       - wait-for-master:
           host: ${MASTER_HOST}
 
+  terminate-gke-cluster:
+    parameters:
+      cluster-id:
+        type: string
+    steps:
+      - run:
+          name: Terminate GKE Cluster
+          when: always
+          command: |
+            gcloud container clusters delete <<parameters.cluster-id>> --quiet --region=us-west1
+
+  setup-gke-cluster:
+    parameters:
+      cluster-id:
+        type: string
+      det-version:
+        type: string
+      cluster-version:
+        type: string
+      machine-type:
+        type: string
+      num-machines:
+        type: integer
+        default: 1
+      gpu-type:
+        type: string
+      gpus-per-machine:
+        type: integer
+        default: 1
+    steps:
+      - set-cluster-id:
+          cluster-id: <<parameters.cluster-id>>
+      - gke/create-cluster:
+          cluster: ${CLUSTER_ID}
+          perform-login: true
+          install-kubectl: true
+          additional-args: |
+            --cluster-version=<<parameters.cluster-version>> \
+            --machine-type=<<parameters.machine-type>> \
+            --num-nodes=<<parameters.num-machines>> \
+            --accelerator type=<<parameters.gpu-type>>,count=<<parameters.gpus-per-machine>> \
+            --region=us-west1 \
```

Comment:
non-blocking: it seems we are hard-coding the region to be `us-west1`. Can we make this configurable and default to us-west1?

Reply:
Sure updated it.